src/unix/unix_c/unix_get_pw_gr_nam_id_job.c

```diff
@@ -81,9 +82,17 @@ static value alloc_group_entry(struct group *entry)
     {                                                                  \
         size_t buffer_size = sysconf(_SC_##CONF##_R_SIZE_MAX);         \
         if (buffer_size == (size_t)-1) buffer_size = 16384;            \
-        job->buffer = (char *)lwt_unix_malloc(buffer_size);            \
-        job->result = FUNC##_r(job->ARG, &job->entry, job->buffer,     \
-                               buffer_size, &job->ptr);                \
+        while (1)                                                   \
+        {                                                              \
+            job->buffer = (char *)lwt_unix_malloc(buffer_size);        \
+            job->result = FUNC##_r(job->ARG, &job->entry, job->buffer, \
+                                   buffer_size, &job->ptr);            \
+            if (job->result == 0) break;                               \
```

Comment:
This check is redundant with the next one, and it's probably best to delete it because the `!= ERANGE` is better for clarity.

Reply:
Fixed.